gateway/ln-gateway/src/lightning/lnd.rs

```diff
@@ -464,77 +474,163 @@ impl ILnRpcClient for GatewayLndClient {
-                    }) {
-                    Ok(Some(payment)) if payment.status() == PaymentStatus::Succeeded => {
-                        info!("LND payment succeeded for invoice {invoice:?}");
-                        break hex::FromHex::from_hex(payment.payment_preimage.as_str()).map_err(
-                            |error| LightningRpcError::FailedPayment {
-                                failure_reason: format!("Failed to convert preimage {error:?}"),
-                            },
-                        )?;
-                    }
-                    Ok(Some(payment)) if payment.status() == PaymentStatus::InFlight => {
-                        debug!("LND payment is inflight");
-                        continue;
-                    }
-                    Ok(Some(payment)) => {
-                        info!("LND payment failed for invoice {invoice:?} with {payment:?}");
-                        let failure_reason = payment.failure_reason();
-                        return Err(LightningRpcError::FailedPayment {
-                            failure_reason: format!("{failure_reason:?}"),
-                        });
-                    }
-                    Ok(None) => {
-                        info!("LND payment failed for invoice {invoice:?} with no payment status");
-                        return Err(LightningRpcError::FailedPayment {
-                            failure_reason: format!(
-                                "Failed to get payment status for payment hash {:?}",
-                                invoice.payment_hash
-                            ),
-                        });
-                    }
+                    .map_err(|status| {
+                        info!("LND payment request failed for invoice {invoice:?} with {status:?}");
+                        LightningRpcError::FailedPayment {
+                            failure_reason: format!("Failed to make outgoing payment {status:?}"),
+                        }
+                    });
+
+                match htlc_attempt.map(Response::into_inner) {
+                    Ok(payment) => match payment.status() {
+                        HtlcStatus::Failed => {
+                            info!("LND payment failed for invoice {invoice:?} with {payment:?}");
+                            if let Some(failure) = payment.failure {
+                                last_failed_reason = Some(failure.code().as_str_name().to_string());
+                            }
+                        }
+                        HtlcStatus::InFlight => {
+                            debug!("LND payment is inflight, tracking...");
+                            let stream = client.router().track_payment_v2(TrackPaymentRequest {
+                                payment_hash: invoice.payment_hash.to_byte_array().to_vec(),
+                                no_inflight_updates: true, // just need payment success/failure
```

Comment:
We had this set to `true` before, might be helpful just to `debug` log the payment for inflight updates?